Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Error.swift

```diff
@@ -80,10 +81,38 @@ extension PathHierarchy.Error {
     ///   - fullNameOfNode: A closure that determines the full name of a node, to be displayed in collision diagnostics to precisely identify symbols and other pages.
     /// - Note: `Replacement`s produced by this function use `SourceLocation`s relative to the link text excluding its surrounding syntax.
     func makeTopicReferenceResolutionErrorInfo(fullNameOfNode: (PathHierarchy.Node) -> String) -> TopicReferenceResolutionErrorInfo {
+        // Both `.unknownDisambiguation(...)` and `.lookupCollisions(...)` create solutions on the same format from the same information.
         // This is defined inline because it captures `fullNameOfNode`.
-        func collisionIsBefore(_ lhs: (node: PathHierarchy.Node, disambiguation: String), _ rhs: (node: PathHierarchy.Node, disambiguation: String)) -> Bool {
-            return fullNameOfNode(lhs.node) + lhs.disambiguation
-                 < fullNameOfNode(rhs.node) + rhs.disambiguation
+        func makeCollisionSolutions(
+            from candidates: [(node: PathHierarchy.Node, disambiguation: String)],
+            nextPathComponent: PathHierarchy.PathComponent,
+            partialResultPrefix: Substring
+        ) -> (
+            pathPrefix: Substring,
+            foundDisambiguation: Substring,
+            solutions: [Solution]
+        ) {
+            let pathPrefix = partialResultPrefix + nextPathComponent.name
+            let foundDisambiguation = nextPathComponent.full.dropFirst(nextPathComponent.name.count)
+            let replacementRange = SourceRange.makeRelativeRange(startColumn: pathPrefix.count, length: foundDisambiguation.count)
+            
+            let solutions: [Solution] = candidates
+                .map { (fullName: fullNameOfNode($0.node), disambiguation: $0.disambiguation) }
+                .sorted { lhs, rhs in
+                    // Sort by name first and disambiguation second
+                    if lhs.fullName == rhs.fullName {
+                        return lhs.disambiguation < rhs.disambiguation
+                    }
+                    return lhs.fullName < rhs.fullName
+                }
+                .map { (fullName: String, suggestedDisambiguation: String) -> Solution in
+                    // In contexts that display the solution message on a single line by removing newlines, this extra whitespace makes it look correct ─────────────╮
```

Comment:
I am not sure I understand what this is referring to.

Reply:
The default diagnostic formatter outputs diagnostic solutions (phrased as "suggestions") on a single line. For example:

```
4 | /// Here's a link that uses the wrong symbol kind disambiguation in a link:
5 + /// - ``myFunction(with:and:)-struct``
  |                              ╰─suggestion: Remove '-struct' for'func myFunction(with one: Int, and two: String) -> Bool'
```

If you look closely at the suggestion text you can see that there's a missing space between "for" and the symbol name.

With these changes this suggestion text isn't missing that space:
```
suggestion: Remove '-struct' for 'func myFunction(with one: Int, and two: String) -> Bool'
                                ^
                                with this change there is an added space here
```